automation/automation/DriverProgram.cs

```diff
@@ -0,0 +1,90 @@
+            
+            //Adding 1st item in the wishlist
+            IWebElement electronics = driver.FindElement(By.XPath("//span[contains(text(),'Electronics')]"));
+            Thread.Sleep(2000);            
+            electronics.Click();        
+            IWebElement eleVivo = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//a[contains(text(),'Vivo')]")));
+            // clicking for product's page
+            eleVivo.Click();
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[1]")));
+            IWebElement addToWishList1 = driver.FindElement(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[2]"));
+            addToWishList1.Click();
+
+            //Adding item 2nd in the wishlist
+            IWebElement eleMen = driver.FindElement(By.XPath("//span[contains(text(),'Men')]"));
+            Thread.Sleep(2000);
+            eleMen.Click();
+            IWebElement eleTrou = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//a[contains(text(),'Casual Trousers')]")));
+            // clicking for product's page
+            eleTrou.Click();
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[3]")));
+            IWebElement addToWishList2 = driver.FindElement(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[3]"));
+            addToWishList2.Click();
+
+            //Adding item 3rd in the wishlist
+            IWebElement eleBabyKids = driver.FindElement(By.XPath("//span[contains(text(),'Baby & Kids')]"));
+            Thread.Sleep(2000);
+            eleBabyKids.Click();
+            IWebElement eleSoftToys = wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementExists(By.XPath("//a[contains(text(),'Soft Toys')]")));
+            // clicking for product's page
+            eleSoftToys.Click();
+            wait.Until(SeleniumExtras.WaitHelpers.ExpectedConditions.ElementIsVisible(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[3]")));
+            IWebElement addToWishList3 = driver.FindElement(By.XPath("(//div/*[namespace-uri()='http://www.w3.org/2000/svg']/*[@fill-rule='evenodd'])[2]"));
+            addToWishList3.Click();
+
+            //Moving on to the user and checking the wishlisted elements count
+            IWebElement user = driver.FindElement(By.XPath("//body/div/div/div/div/div/div[3]/div[1]"));
+            String Check = driver.FindElement(By.XPath("//div[@id='container']//div//div//div//div//div//div//div//div//div//div//div[contains(text(),'2')]")).Text;
+            if (Check == "3")
```

Comment:
Since scenario is about adding multiple items to wishlist, we can add more than 3 products too. This will fail the test when it shouldn't. Better to have check for >3 as the problem statement says "3-4 products".

Reply:
I have made changes keeping this condition